pkg/console/util.go

```diff
@@ -398,6 +410,42 @@ func getRemoteConfig(configURL string) (*config.HarvesterConfig, error) {
 	return harvestCfg, nil
 }
 
+func retryRemoteConfig(configURL string, g *gocui.Gui) (*config.HarvesterConfig, error) {
+	var confData []byte
+	client := newProxyClient()
+
+	printToPanel(g, fmt.Sprintf("Fetching %s...", configURL), installPanel)
+
+	retries := 30
+	interval := 10
+	err := retryOnError(int64(retries), int64(interval), func(remainingRetries int64) error {
+		var e error
+		confData, e = getURL(client, configURL)
+		if e != nil {
+			logrus.Error(e)
+			printToPanel(g, e.Error(), installPanel)
+			printToPanel(g, fmt.Sprintf("Retry after %d seconds (Remaining: %d)...", interval, remainingRetries), installPanel)
+		}
+		return e
+	})
+
+	if err != nil {
+		msg := fmt.Sprintf("Fail to fetch config: %v", err)
+		logrus.Error(msg)
```

Comment:
Log the error in the caller. If you want to add more info just return a wrapped error: fmt.Errorf("Fail to fetch config: %w", err)

Reply:
Move up to the caller.